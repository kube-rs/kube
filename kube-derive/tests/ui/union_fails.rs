use kube_derive::CustomResource;
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};

#[derive(CustomResource, Serialize, Deserialize, JsonSchema)]
union FooSpec {
    int: u32,
}

fn main() {}
