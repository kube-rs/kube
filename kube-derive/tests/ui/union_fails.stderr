error: Unions can not #[derive(CustomResource)]
 --> tests/ui/union_fails.rs:6:7
  |
6 | union FooSpec {
  |       ^^^^^^^

error: Serde does not support derive for unions
 --> tests/ui/union_fails.rs:6:1
  |
6 | / union FooSpec {
7 | |     int: u32,
8 | | }
  | |_^
